### PR TITLE
Feature/add.logging.with.logrus

### PR DESCRIPTION
**Pull request by aleksandrmelnikov**

### CODE REVIEW

File: manager/secret_manager.go [hunk 1]
```diff
@@ -21,14 +27,30 @@ func (r *ResourceManager) CreateSecret(namespace string, secret *model.Secret) (
 func (r *ResourceManager) SecretExists(namespace string, name string) (exists bool, err error) {
 	var foundSecret *model.Secret
 	var statusError *errors.StatusError
+
 	foundSecret, err = r.kubeClient.SecretExists(namespace, name)
 	if err != nil {
 		if goerrors.As(err, &statusError) {
 			if statusError.ErrStatus.Reason == "NotFound" {
+				logging.Logger.Log.WithFields(log.Fields{
```
Comment (rushtehrani) [inline manager/secret_manager.go]: @aleksandrmelnikov I don't think we need to log status error like separately, I think logging just the top most error would be sufficient.

File: manager/secret_manager.go [hunk 2]
```diff
@@ -21,14 +27,30 @@ func (r *ResourceManager) CreateSecret(namespace string, secret *model.Secret) (
...
 				return false, util.NewUserError(codes.NotFound, "Secret Not Found.")
 			}
+			logging.Logger.Log.WithFields(log.Fields{
+				"Namespace": namespace,
+				"Name":      name,
+				"Error":     err.Error(),
+			}).Error("Checking existence of secret.")
 			return false, util.NewUserError(codes.Unknown, "Error when checking existence of secret.")
 		}
+		logging.Logger.Log.WithFields(log.Fields{
```
Comment (rushtehrani) [inline manager/secret_manager.go]: So basically move this up right under `if err !== nil` and log `err` once. Thoughts?

Comment (aleksandrmelnikov) [reply manager/secret_manager.go]: Hmm, I think I see your point. Moving it up right under err !== nil, would capture the error.
The rest if-else checks are for user reporting.

Should this adjustment be applied elsewhere too?

Comment (rushtehrani) [reply manager/secret_manager.go]: That's correct. Yes, we should adjust this everywhere else.

**Comment by rushtehrani** [comment]
@aleksandrmelnikov everything else LGTM, I'll merge once you make the adjustments we discussed.

File: manager/secret_manager.go (hunk 2 above)
Comment (aleksandrmelnikov) [reply manager/secret_manager.go]: Right on, will do

**Comment by rushtehrani** [comment]
@aleksandrmelnikov let's get this refactored so we can merge it before we get more conflicts.

**Comment by aleksandrmelnikov** [comment]
@rushtehrani Refactored per feedback.

**Pull request merged**